modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/selector/attribute/aggregator/incremental/DistinctCountIncrementalAttributeAggregator.java

```diff
@@ -0,0 +1,106 @@
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.ReturnAttribute;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.exception.SiddhiAppRuntimeException;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.expression.Expression;
+
+/**
+ * {@link IncrementalAttributeAggregator} to calculate count based on an event attribute.
+ */
+@Extension(
+        name = "distinctCount",
+        namespace = "incrementalAggregator",
+        description = "Returns the count of all events, in incremental event processing",
+        parameters = {
+                @Parameter(name = "arg",
+                        description = "The attribute that needs to be counted.",
+                        type = {DataType.INT, DataType.LONG, DataType.DOUBLE,
+                                DataType.FLOAT, DataType.STRING, DataType.BOOL})
+        },
+        returnAttributes = @ReturnAttribute(
+                description = "Returns the distinct event count as a long.",
+                type = {DataType.LONG}),
+        examples = @Example(
+                syntax = " define aggregation cseEventAggregation\n from cseEventStream\n" +
+                        " select distinctCount(symbol) as countEvents,\n aggregate by timeStamp every sec ... hour;",
+                description = "distinctCount(symbol) returns the distinct count of all the symbols based on their " +
+                        "arrival and expiry. The count is calculated for sec, min and hour durations."
+        )
+)
+public class DistinctCountIncrementalAttributeAggregator extends IncrementalAttributeAggregator {
+
+    private Attribute[] baseAttributes;
+    private Expression[] baseAttributesInitialValues;
+
+    @Override
+    public void init(String attributeName, Attribute.Type attributeType) {
+        Attribute set;
+        Expression setInitialValue;
+
+        // distinct-count is not supported for object types.
+        if (attributeType.equals(Attribute.Type.FLOAT) || attributeType.equals(Attribute.Type.DOUBLE)
+                || attributeType.equals(Attribute.Type.INT) || attributeType.equals(Attribute.Type.LONG)
+                || attributeType.equals(Attribute.Type.STRING) || attributeType.equals(Attribute.Type.BOOL)) {
+            set = new Attribute("AGG_SET_".concat(attributeName), Attribute.Type.OBJECT);
+            setInitialValue = Expression.function("createSet", Expression.variable(attributeName));
+        } else {
+            throw new SiddhiAppRuntimeException(
+                    "Distinct count aggregation cannot be executed on attribute type " + attributeType.toString());
+        }
+
+        this.baseAttributes = new Attribute[]{set};
+        this.baseAttributesInitialValues = new Expression[]{setInitialValue};
+
+        assert baseAttributes.length == baseAttributesInitialValues.length;
```

Comment:
dont use assert 
check and throw exception

Reply:
Thanks for pointing this out. I referred to an existing incremental aggregator and put this way. 

This check is done in [AggregationParser.java#L563](https://github.com/wso2/siddhi/blob/838737d241a83053f4e7936c256823227d89b60c/modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/parser/AggregationParser.java#L563) hence checking in each and every Attribute Aggregator impl would be unnecessary. As such, removed the assertion in all incremental aggregators in e75340e